lib/api.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  AuthorizationType,
+  FieldLogLevel,
+  GraphqlApi,
+  MappingTemplate,
+  SchemaFile,
+} from "@aws-cdk/aws-appsync-alpha";
+import { AttributeType, BillingMode, Table } from "aws-cdk-lib/aws-dynamodb";
+import { Construct } from "constructs";
+import { join } from "path";
+import { SFNPutNodes } from "./sfn-put-node";
+
+interface CompendiumApiProps {}
+
+export class CompendiumAPI extends Construct {
+  constructor(scope: Construct, id: string, props: CompendiumApiProps) {
+    super(scope, id);
+
+    const dynamoTable = new Table(this, `${id}Table`, {
+      tableName: id,
+      partitionKey: { name: "PK", type: AttributeType.STRING },
+      sortKey: { name: "SK", type: AttributeType.STRING },
+      billingMode: BillingMode.PAY_PER_REQUEST,
+    });
+
+    const sfnPutNodes = new SFNPutNodes(this, id, { dynamoTable });
+
+    dynamoTable.addGlobalSecondaryIndex({
+      indexName: "Inverse",
+      partitionKey: {
+        name: "SK",
+        type: AttributeType.STRING,
+      },
+      sortKey: {
+        name: "PK",
+        type: AttributeType.STRING,
+      },
+    });
+    dynamoTable.addGlobalSecondaryIndex({
+      indexName: "GSI1",
+      partitionKey: {
+        name: "GSI1PK",
+        type: AttributeType.STRING,
+      },
+      sortKey: {
+        name: "GSI1SK",
+        type: AttributeType.STRING,
+      },
+    });
+
+    const api = new GraphqlApi(this, "this", {
```

Comment:
Let's use same naming convention for all cloudformation ids:
```suggestion
    const api = new GraphqlApi(this, `${id}GraphqlApi`, {
```

Please check other resources as well

Reply:
fixed